pdns/rec_channel.cc

```diff
@@ -32,6 +32,32 @@ RecursorControlChannel::~RecursorControlChannel()
     unlink(d_local.sun_path);
 }
 
+static void setSocketBuffer(int fd, int optname, uint32_t size)
+{
+  uint32_t psize=0;
+  socklen_t len=sizeof(psize);
+
+  if (getsockopt(fd, SOL_SOCKET, optname, (void*)&psize, &len))
+    throw PDNSException("Unable to getsocket buffer size: "+stringerror());
+
+  if (psize > size)
+    return;
+
+  if (setsockopt(fd, SOL_SOCKET, optname, (const void*)&size, sizeof(size)) < 0 )
+    throw PDNSException("Unable to raise socket buffer size: "+stringerror());
```

Comment:
Thanks for making this change! I'm not sure we should throw an exception if we can't raise the buffer size, though, because it can occur if the system has a low maximum buffer size, and a lower buffer size might not prevent us from successfully complete the operation while raising this exception will.

Reply:
This is a bit of a hard decision; e.g. on OpenBSD, the buffer sizes are pretty small by default, making e.g. rec_control help fail and possible get-all as well (once the numbers are grown a bit or another stat is added). On the other hand rec itself will probably run fine.  So I'll remove the exception.